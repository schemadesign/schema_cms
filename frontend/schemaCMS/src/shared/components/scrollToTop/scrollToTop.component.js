import { useEffect } from 'react';
import { useLocation } from 'react-router-dom';

export function ScrollToTop({ element = window }) {
  const { pathname } = useLocation();

  useEffect(() => {
    element.scrollTo(0, 0);
  }, [pathname]);

  return null;
}
