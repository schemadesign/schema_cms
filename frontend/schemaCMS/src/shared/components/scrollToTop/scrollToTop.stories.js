import React, { Fragment, PureComponent } from 'react';
import { storiesOf } from '@storybook/react';
import { Router } from 'react-router';
import { createMemoryHistory } from 'history';
import { Button } from 'schemaUI';

import { withRouter, withTheme } from '../../../.storybook/decorators';
import { ScrollToTop } from './scrollToTop.component';

export const defaultProps = {
  history: createMemoryHistory(),
  keyLength: 0,
}

export class Wrapper extends PureComponent {
  state = {
    path: '/init'
  }

  componentDidMount() {
    setTimeout(() => {
      window.scrollTo(0, 1000);
    }, 500);
  }

  handleChangeLocation = () => {
    const id = Math.round(Math.random() * 1000).toString(16);
    const path = `/some-other-url-${id}`;

    this.setState({ path });
    this.props.history.push(path);
  }

  render() {
    return (
      <Router {...this.props}>
        <ScrollToTop />
        <h1>
          Scroll to top
        </h1>
        <h3 style={{backgroundColor: '#777', padding: '3px 15px', display: 'inline-block'}}>
          {this.state.path}
        </h3>
        <p>Lorem ipsum dolor sit amet,</p>
        <p>consectetur adipiscing elit.</p>
        <p>In vel augue eget enim auctor iaculis,</p>
        <p>Donec quis pulvinar urna,</p>
        <p>ac malesuada nisi.</p>
        <p>Cras at lorem auctor,</p>
        <p>Linterdum dolor consequat,</p>
        <p>Lorem ipsum dolor sit amet,</p>
        <p>consectetur adipiscing elit.</p>
        <p>In vel augue eget enim auctor iaculis.</p>
        <p>Donec quis pulvinar urna,</p>
        <p>ac malesuada nisi.</p>
        <p>Cras at lorem auctor.</p>
        <p>Linterdum dolor consequat.</p>
        <Button
          onClick={this.handleChangeLocation}
          customStyles={{padding: '0 20px', margin: '20px 5px'}}
          inverse
        >
          Change location
        </Button>
      </Router>
    );
  }
}

storiesOf('Shared Components|ScrollToTop', module)
  .addDecorator(withRouter)
  .addDecorator(withTheme())
  .add('Default', () => <Wrapper {...defaultProps} />);
