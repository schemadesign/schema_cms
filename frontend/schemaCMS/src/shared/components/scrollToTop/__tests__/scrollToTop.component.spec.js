import React from 'react';
import { shallow } from 'enzyme';
import { Router } from 'react-router';
import { createMemoryHistory } from 'history';

import { ScrollToTop } from '../scrollToTop.component';

describe('ScrollToTop: Component', () => {
  const routerProps = {
    history: createMemoryHistory(),
  };

  const component = props => (
    <Router {...routerProps}>
      <ScrollToTop {...props}/>
    </Router>
  );

  const render = (props = {}) => shallow(component(props));

  it('should render correctly', () => {
    const wrapper = render();

    expect(wrapper).toMatchSnapshot();
  });
});
