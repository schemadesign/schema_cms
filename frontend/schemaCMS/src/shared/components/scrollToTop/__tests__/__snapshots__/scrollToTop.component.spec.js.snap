// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollToTop: Component should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "history": Object {
        "action": "PUSH",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "yrotmq",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          Object {
            "hash": "",
            "key": "l2rz1i",
            "pathname": "/test/2",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 1,
        "length": 2,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "l2rz1i",
          "pathname": "/test/2",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
      "location": Object {
        "hash": "",
        "key": "l2rz1i",
        "pathname": "/test/2",
        "search": "",
        "state": undefined,
      },
      "match": Object {
        "isExact": false,
        "params": Object {},
        "path": "/",
        "url": "/",
      },
      "staticContext": undefined,
    }
  }
>
  <ScrollToTop />
</ContextProvider>
`;
